src/raft.py

```diff
@@ -49,14 +52,24 @@ async def request_leader_based_restructure(self, updated_edges):
             The function returns true when the calling node is the leader.
         '''
         if raftos.get_leader() == self.node_address:
+            
             current_graph = await self.graph.get()
-            # TODO
-            # logic to look at current graph and check if it can be modified,
-            # adding edges to the graph, removing edges should be done/called from here.
-
-            new_edges = {} # REMOVE once TODO above is complete
+        
+            # Adding edges
+            # Check whether edges are not already connected
+            for edge in added_edges:
+                if edge not in current_graph[self.node.node_id]:
+                    graph_funcs.addEdge(current_graph, self.node.node_id, edge)
 
-            await self.graph.update(new_edges)
+            # Deleted edges
+            # Check whether edges are connected
+            for edge in deleted_edges:
+                if edge not in current_graph[self.node.node_id]:
+                    # TODO: Check whether without the edge a flow is still possible??? 
+                    graph_funcs.deleteEdge(current_graph, self.node.node_id, edge)
+                    
+            # TODO: This only displays the dependency graph, i.e. the flow is not being created here. Where should we do that???
```

Comment:
each entry on the adjency list can be a tuple. e.g. 

```python
graph = {
1: set((2, 4), (4, 60))
}
```

here, node 1 has an edge to node 2 and 4. Orders 4 quantities of material from node 2 and 60 quantities of material from node 4.